src/player_activity.cpp

```diff
@@ -240,23 +240,28 @@ void player_activity::do_turn( Character &you )
     }
     // Only do once every two minutes to loosely simulate consume times,
     // the exact amount of time is added correctly below, here we just want to prevent eating something every second
-    if( calendar::once_every( 2_minutes ) && *this && !you.is_npc() && type->valid_auto_needs() &&
-        !you.has_effect( effect_nausea ) ) {
-        if( you.stomach.contains() <= you.stomach.capacity( you ) / 4 && you.get_kcal_percent() < 0.95f &&
-            !no_food_nearby_for_auto_consume ) {
-            int consume_moves = get_auto_consume_moves( you, true );
-            moves_left += consume_moves;
-            if( consume_moves == 0 ) {
-                no_food_nearby_for_auto_consume = true;
+    if( calendar::once_every( 2_minutes ) && *this && !you.is_npc() ) {
+        if( type->valid_auto_needs() && !you.has_effect( effect_nausea ) ) {
+            if( you.stomach.contains() <= you.stomach.capacity( you ) / 4 && you.get_kcal_percent() < 0.95f &&
+                !no_food_nearby_for_auto_consume ) {
+                int consume_moves = get_auto_consume_moves( you, true );
+                moves_left += consume_moves;
+                if( consume_moves == 0 ) {
+                    no_food_nearby_for_auto_consume = true;
+                }
             }
-        }
-        if( you.get_thirst() > 130 && !no_drink_nearby_for_auto_consume ) {
-            int consume_moves = get_auto_consume_moves( you, false );
-            moves_left += consume_moves;
-            if( consume_moves == 0 ) {
-                no_drink_nearby_for_auto_consume = true;
+            if( you.get_thirst() > 130 && !no_drink_nearby_for_auto_consume ) {
+                int consume_moves = get_auto_consume_moves( you, false );
+                moves_left += consume_moves;
+                if( consume_moves == 0 ) {
+                    no_drink_nearby_for_auto_consume = true;
+                }
             }
         }
+        // Also update map and visibility caches every 2 minutes
```

Comment:
Why?

Reply:
Lighting conditions can change while the player is performing an activity, such as a construction that takes place through sunrise.  Now that visibility is no longer calculated every draw, it should be done here for the player to perceive a lighting change during the activity.  Doing so every turn would seem excessive given the speed of activities, so I opted for a calculation every 2 minutes, similar to hunger.

Reply:
I don't think it is a good approach. We can update player hunger every 2 minutes, but it is not good for visibility or lighting. Lighting cache should be updated whenever lighting conditions change, visibility cache should be updated when visibility changes - e.g. when lighting condition change, when player moves, when player constructs something, when fields around player change their state, etc.

Reply:
Can't you get when to update from world time implicating light level change throughout sunset / sunrise?
I don't know if it is a good solution, but it would react to the change directly and should be faster if the brightness doesn't change.
It would still be rebuilding unnecessarily when player is underground and doesn't see natural light.

Reply:
Current plan is to make the conditions more general rather than specific, so mainly position and time.  Time in particular, should cover nearly all the conditions mentioned here.  To keep everything in one place, I would advice that further suggestions be made in #68138.  Thanks.